### PR TITLE
s1227: sudo credentials

### PR DESCRIPTION
**Pull request by rimar-blink**

### CODE REVIEW

File: implementation/cli.go [hunk 1]
```diff
@@ -389,42 +422,57 @@ func executeCoreAzureAction(e *execution.PrivateExecutionEnvironment, ctx *plugi
...
+func initKubernetesEnvironment(e *execution.PrivateExecutionEnvironment, environment []string, bearerToken string, apiServerURL string, verifyCertificate bool) ([]byte, error) {
+
+	output, err := common.ExecuteBash(e, nil, environment, "whoami && pwd && env")
+	if err != nil {
+		return output, err
+	}
+	log.Infof("whoami && pwd && env output: %s", output)
 
-	cmd := fmt.Sprintf("kubectl config set-cluster cluster --server=%s", apiServerURL)
+	cmd := fmt.Sprintf("/opt/blink/kubectl config set-cluster cluster --server=%s", apiServerURL)
```
Comment (itsofirblink) [inline implementation/cli.go:455]: const `/opt/blink`?

**Review by itsofirblink — APPROVED** [review]

File: implementation/cli.go [hunk 2]
```diff
@@ -86,30 +112,120 @@ func executeCoreAWSAction(e *execution.PrivateExecutionEnvironment, ctx *plugin.
...
 
 func executeCoreGITAction(e *execution.PrivateExecutionEnvironment, ctx *plugin.ActionContext, request *plugin.ExecuteActionRequest) ([]byte, error) {
-	credentials, err := ctx.GetCredentials("github")
+	sshCredentials, _ := ctx.GetCredentials("ssh")
+	basicAuthCredentials, _ := ctx.GetCredentials("github")
+	basicAuthType := "github"
+	if basicAuthCredentials == nil {
+		basicAuthCredentials, _ = ctx.GetCredentials("gitlab")
+		basicAuthType = "gitlab"
+	}
```
Comment (itsofirblink) [inline implementation/cli.go:121]: that's a problem cause you may pass both `github` and `gitlab` and it will not work :(

**Review by itsofirblink — CHANGES_REQUESTED** [review]

File: implementation/cli.go (hunk 2 above)
Comment (rimar-blink) [reply implementation/cli.go:121]: you can't pass more than 1 credentials to a step

File: implementation/cli.go [hunk 3]
```diff
@@ -35,40 +33,71 @@ const (
...
+	lines = append(lines, fmt.Sprintf("%s = %v\n", "region", region))
+	awsCredFileContent := strings.Join(lines, "\n")
+
+	if err = cliUserPee.CreateDirectory(path.Join(cliUserPee.GetHomeDirectory(), ".aws")); err != nil {
+		return nil, errors.Wrap(err, "failed to create .aws directory")
+	}
+
+	if err = cliUserPee.WriteToFile(path.Join(cliUserPee.GetHomeDirectory(), ".aws", "credentials"), []byte(awsCredFileContent), 0600); err != nil {
+		return nil, errors.Wrap(err, "failed to write to .aws/credentials")
+	}
```
Comment (itsofirblink) [inline implementation/cli.go:88]: move to a function and call only if `m` is not empty.

File: implementation/cli.go [hunk 4]
```diff
@@ -35,40 +33,71 @@ const (
...
 	}
 
+	var m map[string]string
 	credentials, err := ctx.GetCredentials("aws")
+	// if no credentials provided, execute without credentials, otherwise resolve assumed role etc.
 	if err == nil {
-		awsEnvironmentVariables, err := resolveAWSCredentials(credentials, region)
-		if err != nil {
+		if m, err = resolveAwsCreds(m, credentials, region); err != nil {
 			return nil, err
```
Comment (itsofirblink) [inline implementation/cli.go:61]: don't return error here, it may be a failure cause of missing secrets, just continue to execute without credentials :)

**Review by itsofirblink — CHANGES_REQUESTED** [review]

File: implementation/cli.go (hunk 4 above)
Comment (rimar-blink) [reply implementation/cli.go:61]: ok

File: implementation/cli.go (hunk 3 above)
Comment (rimar-blink) [reply implementation/cli.go:88]: moving to a function, but i want it to run even if m is null, to set the region

**Review by itsofirblink — APPROVED** [review]

**Pull request merged**